### PR TITLE
Build on OSX

### PR DESCRIPTION
**Pull request by warmenhoven**
This gets mupen64plus_next running on OSX, both on x64 and arm64.

### CODE REVIEW

File: mupen64plus-rsp-paraLLEl/jit_allocator.cpp [hunk 1]
```diff
@@ -31,7 +32,12 @@ Allocator::~Allocator()
 
 static size_t align_page(size_t offset)
 {
-	return (offset + 4095) & ~size_t(4095);
+#ifdef _WIN32
+    size_t pagesize = 4095;
+#else
+    size_t pagesize = sysconf(_SC_PAGESIZE) - 1;
```
Comment (m4xw) [inline mupen64plus-rsp-paraLLEl/jit_allocator.cpp:38]: sysconf isn't available on all systems, for libnx i provide it myself from inside RA.
Can't rely on that for the core tho

**Review by m4xw — CHANGES_REQUESTED** [review]

**Comment by warmenhoven** [comment]
Something like this? https://github.com/Themaister/parallel-rsp/pull/5 (This is closer to what flycast does as well.)

**Comment by m4xw** [comment]
I still need to do some local test, i think the ASSIGN_PROC_ADR change is sane to do on all platforms either way.
Other than that it looks acceptable, will get to it the next time i touch it (sometime this week)

**Comment by warmenhoven** [comment]
Thanks! Yes, the `ASSIGN_PROC_ADR` change should be safe on all platforms. It could potentially mask problems, so I wasn't sure if it would be preferable to limit the cast to just Mac. I can change it so it applies to all platforms.

**Comment by warmenhoven** [comment]
I've changed `ASSIGN_PROC_ADR` so it applies to all platforms instead of just osx, and I've rebased onto the current `develop` branch.

**Comment by m4xw** [comment]
That reminds me, subrepo has a incorrect parent reference. Can u just squash out the parallel rsp change and i will just pull it on my end after merge?

**Comment by warmenhoven** [comment]
Yep, I've removed the parallel-rsp change. The osx build might fail until the subrepo is pulled though as rsp_jit.cpp did have a cast fix.

**Comment by m4xw** [comment]
I still see the mupen64plus-rsp-paraLLEl/jit_allocator.cpp change.
If its upstream then i dont need any change for the folder

**Comment by warmenhoven** [comment]
Yep, sorry, missed that. Fixed now. Thanks!

**Comment by m4xw** [comment]
Actually i forgot I dont have my own remote there.... subrepo was fine :P 
I'd prefer to not resolve that conflict in the future tho so will just force clone in the future but whatever for now

**Comment by m4xw** [comment]
https://git.libretro.com/libretro/mupen64plus-libretro-nx/-/jobs/3068798 arm64 is faulty

**Comment by m4xw** [comment]
did a short change, must be arch detection being faulty for apple cross comp

**Pull request merged**